commit eb418c3670c37daf053be72f3a131b6c72446cdd
Author: Jim Klimov <jim@jimklimov.com>
Date:   Fri May 4 22:18:46 2018 +0200

    Problem : @exec_prefix@ expands to ${prefix} literally

OI note: PR is submitted, this patch-file will be not needed for releases
newer than znapzend-0.18.0

diff --git a/configure.ac b/configure.ac
index eb5ea42..aa3e25e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -128,6 +128,62 @@ AC_MSG_RESULT(way to expensive!)
 AC_ARG_VAR(PERL5LIB,   [Colon separated list of perl library directories])
 AC_SUBST(PERL5LIB)
 
+dnl expand ${prefix} and write it out
+conftemp="${prefix}"
+eval conftemp=\"${conftemp}\"
+eval conftemp=\"${conftemp}\"
+PREFIX=${conftemp}
+AC_DEFINE_UNQUOTED(PREFIX, "${conftemp}", [Default base path for product files])
+AC_SUBST(PREFIX)
+
+dnl same for exec_prefix
+conftemp="${exec_prefix}"
+eval conftemp=\"${conftemp}\"
+eval conftemp=\"${conftemp}\"
+EXEC_PREFIX=${conftemp}
+AC_DEFINE_UNQUOTED(EXEC_PREFIX, "${conftemp}", [Default base path for product files])
+AC_SUBST(EXEC_PREFIX)
+
+dnl same for datadir
+conftemp="${datadir}"
+eval conftemp=\"${conftemp}\"
+eval conftemp=\"${conftemp}\"
+DATADIR=${conftemp}
+AC_DEFINE_UNQUOTED(DATADIR, "${conftemp}", [Default path for data files])
+AC_SUBST(DATADIR)
+
+dnl same for sysconfdir
+conftemp="${sysconfdir}"
+eval conftemp=\"${conftemp}\"
+eval conftemp=\"${conftemp}\"
+SYSCONFDIR=${conftemp}
+AC_DEFINE_UNQUOTED(SYSCONFDIR, "${conftemp}", [Default path for configuration files])
+AC_SUBST(SYSCONFDIR)
+
+dnl same for bindir
+conftemp="${bindir}"
+eval conftemp=\"${conftemp}\"
+eval conftemp=\"${conftemp}\"
+BINDIR=${conftemp}
+AC_DEFINE_UNQUOTED(BINDIR, "${conftemp}", [Default path for user executables])
+AC_SUBST(BINDIR)
+
+dnl same for sbindir
+conftemp="${sbindir}"
+eval conftemp=\"${conftemp}\"
+eval conftemp=\"${conftemp}\"
+SBINDIR=${conftemp}
+AC_DEFINE_UNQUOTED(SBINDIR, "${conftemp}", [Default path for system executables])
+AC_SUBST(SBINDIR)
+
+dnl same for libdir
+conftemp="${libdir}"
+eval conftemp=\"${conftemp}\"
+eval conftemp=\"${conftemp}\"
+LIBDIR=${conftemp}
+AC_DEFINE_UNQUOTED(LIBDIR, "${conftemp}", [Default path for system libraries])
+AC_SUBST(LIBDIR)
+
 AC_CONFIG_FILES([
     Makefile
     thirdparty/Makefile
diff --git a/init/znapzend.xml.in b/init/znapzend.xml.in
index b3ec8f9..be28f4b 100644
--- a/init/znapzend.xml.in
+++ b/init/znapzend.xml.in
@@ -28,7 +28,7 @@
         <exec_method 
                 type="method" 
                 name="start" 
-                exec="@exec_prefix@/bin/znapzend --daemonize --pidfile=/dev/null"
+                exec="@BINDIR@/znapzend --daemonize --pidfile=/dev/null"
                 timeout_seconds="180" />
 
         <exec_method
